src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -204,14 +208,50 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
 		{
 			cancellation.ThrowIfCancellationRequested();
 
-			if (settings.SuppressionMechanismEnabled && Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true)
+			if (settings.SuppressionMechanismEnabled && 
+			    (Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true ||
+			    CheckSuppressedComment(diagnostic, cancellation)))
 			{
 				return;
 			}
 
 			reportDiagnostic(diagnostic);
 		}
 
+		private static bool CheckSuppressedComment(Diagnostic diagnostic, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+
+			SyntaxNode root = diagnostic.Location.SourceTree?.GetRoot();
+			SyntaxNode node = root?.FindNode(diagnostic.Location.SourceSpan);
+			bool containsComment = false;
+
+			// Climb to the hill. Looking for comment on parents nodes.
+
+			while (node != null && node != root)
+			{
+				containsComment = CheckSuppressionCommentOnNode(diagnostic, diagnostic.Descriptor.CustomTags.FirstOrDefault(), node, cancellation);
```

Comment:
@saloponov you can put calculation `diagnostic.Descriptor.CustomTags.FirstOrDefault()` out of the `while` - there is no reason to re-calculate it.